doc/man3/OSSL_CMP_log_open.pod

```diff
@@ -0,0 +1,119 @@
+=pod
+
+=head1 NAME
+
+OSSL_CMP_log_open,
+OSSL_CMP_log_close,
+OSSL_CMP_puts,
+OSSL_CMP_log_printf,
+OSSL_CMP_add_error_txt,
+OSSL_CMP_add_error_data,
+OSSL_CMP_add_error_line,
+OSSL_CMP_print_errors_cb
+- functions for logging and error reporting
+
+=head1 SYNOPSIS
+
+ #include <openssl/cmp_util.h>
+
+ typedef enum {OSSL_LOG_EMERG, OSSL_LOG_ALERT, OSSL_LOG_CRIT, OSSL_LOG_ERR,
+               OSSL_LOG_WARNING, OSSL_LOG_NOTICE, OSSL_LOG_INFO, OSSL_LOG_DEBUG}
+    OSSL_CMP_severity;
+ /* declarations resemble those from bio/bss_log.c and syslog.h */
+ typedef int (*OSSL_cmp_log_cb_t) (const char *component,
+                                   const char *file, int lineno,
+                                   OSSL_CMP_severity level, const char *msg);
+
+ int  OSSL_CMP_log_open(void);
+ void OSSL_CMP_log_close(void);
+ int OSSL_CMP_puts(const char *component, const char *file, int lineno,
+                   OSSL_CMP_severity level, const char *msg);
+ int OSSL_CMP_log_printf(OSSL_cmp_log_cb_t log_fn,
+                         const char *func, const char *file, int lineno,
+                         OSSL_CMP_severity level, const char *fmt, va_list argp);
+
+ void OSSL_CMP_add_error_txt(const char *separator, const char *txt);
+ #define OSSL_CMP_add_error_data(txt) OSSL_CMP_add_error_txt(" : ", txt)
+ #define OSSL_CMP_add_error_line(txt) OSSL_CMP_add_error_txt("\n", txt)
+ void OSSL_CMP_print_errors_cb(OSSL_cmp_log_cb_t log_fn);
```

Comment:
As noted previously we should look at the trace API instead of this.

Reply:
The first three of these are helpers not for logging/tracing, but for extending the OpenSSL error queue.
The fourth function meanwhile makes use of the trace API; please see also my above responses.

Reply:
We should not be adding error, logging or tracing functions to the CMP code. Amend the error or tracing code instead as required (in a separate PR). CMP is not about error, logging or tracing - it just uses those utilities to do its job. We should have common functions throughout the codebase and not have the "CMP specific flavour of error functions" or the "CMP specific flavour of tracing functions".